### PR TITLE
Fixing warnings

### PR DESCRIPTION
**Pull request by jleandroperez**
### Fix
In this PR we're fixing a number of Xcode build warnings.

Proper fix for the `Note.h` warning will come... hopefully as soon as Simperium's next steps are outlined.

cc @bummytime (Thanks in advance sir!!)

### Test
- [x] Verify the app builds correctly

### Release
These changes do not require release notes.

### CODE REVIEW

File: Simplenote/Classes/Note.h [hunk 1]
```diff
@@ -48,7 +48,27 @@
...
+//  property name, while exposing a different property setter / getter, and thus, avoiding the collision.
+//
+// In This thermonuclear massive super workaround, we're simply silencing the warning.
+//
+// Proper course of action should be taken as soon as the next steps for Simperium are outlined.
+//
+// TODO: JLP Dec.3.2019.
+//
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wproperty-attribute-mismatch"
```
Comment (bummytime) [inline Simplenote/Classes/Note.h]: For the love of god...thank you @jleandroperez! I agree it's not ideal, but we can make sure this is updated when Simperium is overhauled.

**Comment by jleandroperez** [comment]
Wooo!!! thank you Bummy!!

File: Simplenote/Classes/Note.h [hunk 2]
```diff
@@ -48,7 +48,27 @@
...
+//  -   Eventually CLANG enhanced checks allowed us to notice there's a collision
+//
+//  Proper fix involves a heavy modification in Simperium, which would allow us to keep the `deleted` "internal"
+//  property name, while exposing a different property setter / getter, and thus, avoiding the collision.
+//
+// In This thermonuclear massive super workaround, we're simply silencing the warning.
+//
+// Proper course of action should be taken as soon as the next steps for Simperium are outlined.
+//
+// TODO: JLP Dec.3.2019.
```
Comment (dmsnell) [inline Simplenote/Classes/Note.h]: 👍 also, I know where you live, and I will find you

Comment (jleandroperez) [reply Simplenote/Classes/Note.h]: yyyyeah I knowwwwww 😂

**Pull request merged**